src/openloco/windows/IndustryWindow.cpp

```diff
@@ -0,0 +1,1064 @@
+        }
+
+        if (window == nullptr)
+        {
+            // 0x00456DBC start
+            const uint32_t newFlags = window_flags::flag_8 | window_flags::resizable;
+            window = WindowManager::createWindow(WindowType::industry, industry::windowSize, newFlags, &industry::events);
+            window->number = industryId;
+            window->min_width = 192;
+            window->min_height = 137;
+            window->max_width = 600;
+            window->max_height = 440;
+
+            auto skin = objectmgr::get<interface_skin_object>();
+            if (skin != nullptr)
+            {
+                window->colours[0] = skin->colour_0B;
+                window->colours[1] = skin->colour_0C;
+            }
+            // 0x00456DBC end
+
+            window->saved_view.clear();
+        }
+
+        // TODO: only needs to be called once.
+        common::initEvents();
+
+        window->current_tab = common::widx::tab_industry - common::widx::tab_industry;
+        window->invalidate();
+
+        window->widgets = industry::widgets;
+        window->enabled_widgets = industry::enabledWidgets;
+        window->holdable_widgets = 0;
+        window->event_handlers = &industry::events;
+        window->activated_widgets = 0;
+        auto industryObj = objectmgr::get<industry_object>(industrymgr::get(window->number)->object_id);
+        auto disabledWidgets = 0;
+
+        if (industryObj->produced_cargo[0] == 0xFF)
+            disabledWidgets |= 0x20;
+        if (industryObj->produced_cargo[1] == 0xFF)
+            disabledWidgets |= 0x40;
+
+        window->disabled_widgets = disabledWidgets;
+        window->init_scroll_widgets();
+        industry::initViewport(window);
+
+        return window;
+    }
+
+    namespace production
+    {
+
+        static const gfx::ui_size_t minWindowSize = { 299, 282 };
+
+        static const gfx::ui_size_t maxWindowSize = { 299, 337 };
+
+        static window_event_list events;
+
+        // 0x00455FD9
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            common::repositionTabs(self);
+        }
+
+        // 0x00456079
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            // Draw Units of Cargo sub title
+            auto industry = industrymgr::get(self->number);
+            auto industryObj = objectmgr::get<industry_object>(industry->object_id);
+            auto cargoObj = objectmgr::get<cargo_object>((uint8_t)industryObj->produced_cargo[0]);
+
+            auto args = FormatArguments();
+            args.push(cargoObj->units_and_cargo_name);
+
+            int16_t x = self->x + 2;
+            int16_t y = self->y - 24 + 68;
+
+            gfx::draw_string_494B3F(*dpi, x, y, colour::black, string_ids::production_graph_label, &args);
```

Comment:
Put all of this  (L387 - L398) inside of curly braces so that you can reuse args and x, y variable names